model/src/main/kotlin/Vulnerability.kt

```diff
@@ -19,18 +19,28 @@
 
 package org.ossreviewtoolkit.model
 
+import java.net.URL
+
 /**
  * Base model for all vulnerability providers supported by the advisor.
  */
-interface Vulnerability {
+sealed class Vulnerability {
     /**
      * The ID of a vulnerability. Most likely a CVE identifier.
      */
-    val id: String
+    abstract val id: String
 
     /**
      * Severity of the vulnerability. Most likely expressed using the CVSS.
      * See: https://www.first.org/cvss/
      */
-    val severity: Float
+    abstract val severity: Float
 }
+
+data class NexusIqVulnerability(
```

Comment:
I don't believe that the model should depend on a Nexus IQ-specific class. It should be the opposite: The Nexus IQ implementation should depend on the model.